### PR TITLE
Issue 89: Migrating gov-uk files from s3 and processing through k8-test-data file processor.

### PR DESCRIPTION
**Pull request by nirvishek**
Process all gov_UK files from S3 bucket to generate metadata, rebuild XML and virustotal report.

### CODE REVIEW

File: gov_uk_file_migration/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Gov-UK files migrator and processor
+
+## Usage
+* Set .env variables from .env.sample
+
+## Build
+
+    `To build the Docker image and name it `multilevel-compression`
```
Comment (susilupwork) [inline gov_uk_file_migration/README.md:8]: Why the name is multilevel

File: gov_uk_file_migration/main.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        self.download_dir = Config.LOCAL_REPO_PATH
+
+    def get_files(self):
+
+        # number of files to download
+        n_files_to_fetch = 100  # to be changed to float('inf') to fetch all files
+
+        # download files from s3 bucket using s3client
+        try:
+            list_files = self.s3.download_subdirectory_files(Config.S3_BUCKET, Config.S3_SUB_FOLDER_PREFIX,
```
Comment (susilupwork) [inline gov_uk_file_migration/main.py:28]: You are downloading all files in one go. It is not scalable. You should have taken list of files, download one  and then process. Then jump to next file.

File: gov_uk_file_migration/main.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+            # 2. extract meta data
+            bucket_name = GovUKFileMigration.get_bucket_name(file)
+            file_name = file.split('/')[-1]
+
+            logger.info("GovUKFileMigration::preprocess_files Iterating file: %s |"
+                        " Bucket Name: %s | Filename: %s" % (file, bucket_name, file_name))
+
+            try:
+                # 3. upload the file to minio before passing it to processor.
```
Comment (susilupwork) [inline gov_uk_file_migration/main.py:69]: Good commenting style.

File: gov_uk_file_migration/src/utils/s3_client.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
+import os
```
Comment (susilupwork) [inline gov_uk_file_migration/src/utils/s3_client.py:1]: Do not copy paste s3_client code for every microservice. Put it in gw.testdata.util  or some place. Use it as a library.

**Review by susilupwork — CHANGES_REQUESTED** [review]
- Good commenting style. And Exception handling done. 
- You are downloading all files in one go. It is not scalable. You should have taken list of files, download one  and then process. Then jump to next file.
- Do not copy paste s3_client code for every microservice. Put it in gw.testdata.util  or some place. Use it as a library.  Or you need extra method update in library.

File: gov_uk_file_migration/README.md (hunk 1 above)
Comment (nirvishek) [reply gov_uk_file_migration/README.md]: I copied the readme from my previous module, updated the code now.

File: gov_uk_file_migration/main.py (hunk 2 above)
Comment (nirvishek) [reply gov_uk_file_migration/main.py]: I have refactored the code to fetch file iteratively one at a time with generator method, please review.

File: gov_uk_file_migration/src/utils/s3_client.py (hunk 4 above)
Comment (nirvishek) [reply gov_uk_file_migration/src/utils/s3_client.py]: I have moved the s3_client to storage service, and now calling its endpoint for s3 services.

**Review by susilupwork — APPROVED** [review]
Good to go.

**Pull request merged**